src/internal/kopia/upload.go

```diff
@@ -334,27 +361,43 @@ func streamBaseEntries(
 			return nil
 		}
 
-		if err := cb(ctx, entry); err != nil {
-			entName, err := decodeElement(entry.Name())
-			if err != nil {
-				entName = entry.Name()
-			}
+		entName, err := decodeElement(entry.Name())
+		if err != nil {
+			return errors.Wrapf(err, "unable to decode entry name %s", entry.Name())
+		}
 
-			return errors.Wrapf(err, "executing callback on item %q", entName)
+		// For now assuming that item IDs don't need escaping.
+		itemPath, err := curPath.Append(entName, true)
+		if err != nil {
+			return errors.Wrap(err, "getting full item path for base entry")
 		}
 
-		return nil
-	})
-	if err != nil {
-		name, err := decodeElement(dir.Name())
+		// We need the previous path so we can find this item in the base snapshot's
+		// backup details. If the item moved and we had only the new path, we'd be
+		// unable to find it in the old backup details because we wouldn't know what
+		// to look for.
+		prevItemPath, err := prevPath.Append(entName, true)
 		if err != nil {
-			name = dir.Name()
+			return errors.Wrap(err, "getting previous full item path for base entry")
+		}
+
+		// All items have item info in the base backup. However, we need to make
+		// sure we have enough metadata to find those entries. Ensure that by adding
```

Comment:
Nit: "ensure that..." can be read as a command, rather than a statement of fact.
```suggestion
		// sure we have enough metadata to find those entries. To do that we add
```